Version 0.6.0

doloboyz opened the pull request:
- Dependency updates
- added jsdoc comments to all functions

doloboyz committed:
Version 0.6.0

- Dependency updates
- added jsdoc comments to all functions

doloboyz committed:
Update node version

doloboyz commented:
Not too sure as to why codesandbox CI is failing. Seems to be building successfully.

doloboyz committed:
Added @babel/core as dev dependency

dai-shi commented:
Thanks for the suggestion. Don't worry about codesandbox. 

Do you think you can work to follow https://github.com/jotaijs/jotai-immer/pull/9 ?

doloboyz committed:
Module-first setup attempt

doloboyz commented:
Not sure if I understood correctly, but I made my best attempt at making repo compliant with the above provided packages. Unfortunately, it was riddled with some module resolution errors, hopefully you can provide a quick fix to this.

dai-shi commented on tests/01_basic_spec.tsx:
```diff
@@ -0,0 +1,11 @@
+    expect(src.atomWithMutation).toBeDefined();
+    expect(src.atomWithSubscription).toBeDefined();
+  });
+});
```
We need to change the file name to `01_basic.spec.tsx`.

dai-shi commented:
Can you run `npx prettier -w .`?

doloboyz commented:
Seems like no changes after running npx prettier -w

doloboyz committed:
Ran prettier on the directory

doloboyz commented:
Let me know if you need anything else done

dai-shi commented:
I'll check it later. Do you have anything else you can do?

doloboyz committed:
More fixes

dai-shi committed:
fix tests

dai-shi commented:
@doloboyz Hi, I added some fixes, but haven't tested if examples run correctly.
Also, please check which style of relay.config.js is better.

doloboyz commented:
So unfortunately, I don't believe module-first will work with the tsconfig files in the example subdirectories. I think commonjs needs to be used. If you want me to push the changes, I've implemented fixes for the examples.

doloboyz commented:
also after i adjusted the tsconfigs and package.json to use commonjs rather than module for the sake of the relay compiler, examples fail with 404

doloboyz committed:
Changes to fix examples

- Used commonjs module resolution in tsconfig.json
- Changed 'AppCountriesQuery' to 'appCountriesQuery' in app.tsx
to satisfy the relay compiler
- Changed 'type' in package.json to 'commonjs' to satisfy relay

dai-shi commented:
> So unfortunately, I don't believe module-first will work

It's something I did in other repos, but I don't strongly suggest to do the same here.
So, you can do what you think is the best in this repo.

Also, as nobody else is maintaining this repo. I can invite you to be a maintainer of this repo.
